spec/service_worker/index.html

```diff
@@ -1372,6 +1373,48 @@ <h2 id="status">Status of This Document</h2>
+
+      <p><dfn id="install-event-register-foreign-fetch-scopes-method"><code>registerForeignFetchScopes(<var>subScopes</var>)</code></dfn> method <em class="rfc2119" title="MUST">must</em> run these steps or their <a href="#dfn-processing-equivalence">equivalent</a>:</p>
+      <spec-algorithm>
+      <ol>
+        <li>If the <a href="https://dom.spec.whatwg.org/#dispatch-flag">dispatch flag</a> is unset, then:
+          <ol>
+            <li><a href="http://heycam.github.io/webidl/#dfn-throw">Throw</a> an "<code><a href="http://heycam.github.io/webidl/#invalidstateerror">InvalidStateError</a></code>" exception.</li>
+            <li>Abort these steps.</li>
+          </ol>
+        </li>
+        <li>Let <var>scopeString</var> be the <a href="https://html.spec.whatwg.org/multipage/webappapis.html#incumbent-settings-object">incumbent settings object</a>'s <a href="https://html.spec.whatwg.org/multipage/webappapis.html#global-object">global object</a>'s <a href="#dfn-service-worker-global-scope-service-worker">service worker</a>'s <a href="#dfn-containing-service-worker-registration">containing service worker registration</a>'s <a href="#dfn-scope-url">scope url</a>, <a href="https://url.spec.whatwg.org/#concept-url-serializer">serialized</a>.</li>
+        <li>Let <var>subScopeURLs</var> be an empty list of <a href="https://url.spec.whatwg.org/#concept-url">URLs</a>.</li>
+        <li>For each <var>subScope</var> in <var>subScopes</var>:
+          <ol>
+            <li>Let <var>subScopeURL</var> be the result of <a href="https://url.spec.whatwg.org/#concept-url-parser">parsing</a> <var>subScope</var> with <a href="https://html.spec.whatwg.org/multipage/webappapis.html#entry-settings-object">entry settings object</a>'s <a href="https://html.spec.whatwg.org/multipage/webappapis.html#api-base-url">API base URL</a>.</li>
+            <li>If <var>subScopeURL</var> is failure, throw a <code>TypeError</code> and abort these steps.</li>
+            <li>Let <var>subScopeString</var> be the <a href="https://url.spec.whatwg.org/#concept-url-serializer">serialized</a> <var>subScopeURL</var>.</li>
+            <li>If <var>subScopeString</var> does not start with <var>scopeString</var>, throw a <code>TypeError</code> and abort these steps.</li>
+            <li>Add <var>subScopeURL</var> to <var>subScopeURLs</var>.</li>
+          </ol>
+        </li>
+        <li>Set this <a href="#dfn-service-worker">service worker</a>'s <a href="#dfn-foreign-fetch-scopes">list of foreign fetch scopes</a> to <var>subScopeURLs</var>.</li>
+      </ol>
+      </spec-algorithm>
+
+      <p class="fixme">Maybe subscopes should be parsed using the registrations scope URL instead of the workers API base url. Although that might actually be more surprising/confusing.</p>
```

Comment:
+1 to using settings object's API base URL. Having a consistency with other APIs seems better for devs so they don't have to look up a document for this usage.